backend/src/util/virtualLinkTransaction.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Transaction } from '@model/Transaction'
+import { SaveOptions, RemoveOptions } from '@dbTools/typeorm'
+import { Transaction as dbTransaction } from '@entity/Transaction'
+import { TransactionTypeId } from '@enum/TransactionTypeId'
+import { User } from '@model/User'
+import Decimal from 'decimal.js-light'
+
+const virtualLinkTransaction = (
+  balance: Decimal,
+  amount: Decimal,
+  holdAvailableAmount: Decimal,
+  decay: Decimal,
+  createdAt: Date,
+  validUntil: Date,
+  user: User,
+): Transaction => {
+  const linkDbTransaction: dbTransaction = {
+    id: -2,
```

Comment:
use here -1 and -2 on the first, since in my mind the decay comes first in the list and then the pending link-transactions. But I am open to discuss what the order should be. If we do it like this:

- virtual decay
- virtual pending links transaction
- transactions
- ...

We should reflect the order in the ids

Reply:
In the most cases we will not have virtual pending link transaction. Then we would have -2 as the first id as we always have a decay.